flow/raster_cache.cc

```diff
@@ -77,6 +77,11 @@ RasterCacheResult RasterizePicture(SkPicture* picture,
 
   const SkVector3& scale = matrix.scale();
 
+  if (scale.x() == 0 || scale.y() == 0) {
```

Comment:
This check is redundant as the decomposition would have failed already.

Reply:
Ok, I'll replace it with DCHECK and comment that the decomposition has already taken care of that.